independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -562,12 +567,12 @@ static ScopeInfo getValidScope(Collection<ScopeInfo> stereotypeScopes, Annotatio
     private List<BeanInfo> findBeans(Collection<DotName> beanDefiningAnnotations, List<ObserverInfo> observers,
             List<InjectionPointInfo> injectionPoints, boolean jtaCapabilities) {
 
-        Set<ClassInfo> beanClasses = new HashSet<>();
-        Set<MethodInfo> producerMethods = new HashSet<>();
-        Set<MethodInfo> disposerMethods = new HashSet<>();
-        Set<FieldInfo> producerFields = new HashSet<>();
-        Map<MethodInfo, Set<ClassInfo>> syncObserverMethods = new HashMap<>();
-        Map<MethodInfo, Set<ClassInfo>> asyncObserverMethods = new HashMap<>();
+        Set<ClassInfo> beanClasses = new TreeSet<>();
```

Comment:
Why sometimes `TreeSet` and sometimes `LinkedHashSet`?

Reply:
Only linked collections are used in f8b3805247e0c5b24e1f2d1ef29fdfdf7046a09e